source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,253 @@
+
+   .. tab:: Synchronous
+      :tabid: find-one-sync
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = _restaurantsCollection.Find(filter).FirstOrDefault();
+
+.. tip:: First Document
+
+   The ``FirstOrDefault()`` method returns the first document in 
+   :manual:`natural order </reference/glossary/#std-term-natural-order>`
+   on disk if no sort criteria is specified.
+
+To see a full example of using the ``Find()`` method to find a single document, see
+:ref:`csharp-retrieve-additional-information`.
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To find multiple documents in a collection, pass a query filter to the ``Find()``
+method that specifies the criteria of the documents you want to retrieve.
+
+You can use a **cursor** to iterate over the documents returned by the ``Find()``
+method. A cursor is a mechanism that allows an application to iterate over database 
+results while holding only a subset of them in memory at a given time. Cursors
+are useful when your ``Find()`` method returns a large amount of documents.
+
+To use a cursor to iterate over the documents, pass a 
+query filter to the ``Find()`` method that specifies the criteria of the documents 
+you want to find, then chain the ``ToCursor()`` or ``ToCursorAsync()`` method as follows:
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-cursor-async
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToCursorAsync();
+
+   .. tab:: Synchronous
+      :tabid: find-cursor-sync
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = _restaurantsCollection.Find(filter).ToCursor();
+
+If you are returning a small number of documents, or need your results returned
+as a ``List`` object, use the ``ToList()`` or ``ToListAsync()`` methods.
+
+To find multiple documents in a collection and hold them in memory as a list, pass a query filter 
+to the ``Find()`` method that specifies the criteria of the documents you want 
+to find, then chain the ``ToList()`` or ``ToListAsync()`` method as follows:
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-list-async
+
+      .. code-block:: csharp
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToList();
+
+   .. tab:: Synchronous
+      :tabid: find-list-sync
+
+      .. code-block:: csharp
+
+         var restaurants = _restaurantsCollection.Find(filter).ToListAsync();
+
+To see a full example of using the ``Find()`` method to find multiple documents,
+see :ref:`csharp-retrieve-additional-information`.
+
+Find All Documents
+~~~~~~~~~~~~~~~~~~
+
+To find all documents in a collection, pass an empty
+``BsonDocument`` as a query filter. Then chain the ``ToList()`` or ``ToListAsync()``
+method to return the documents as a ``List`` object, or the ``ToCursor()`` or ``ToCursorAsync()``
+method to return the documents as a cursor.
```

Comment:
Question/Possible Issue: Any reason ``ToList()`` is presented first here? I think trying to hold an entire collection in a list may not be a good thing to implicitly advocate for.

Suggestion: This section is short, and I think the second sentence of the first paragraph largely duplicates the preceding subsection and can be deleted. I think the following edits could be beneficial:

- Remove the Find All documents section and move the content to an admonition
- Place the admonition in the "Find Multiple Documents" section
- Shorten the content as follows:

```
.. note: Find All Documents

   To find all documents in a collection, pass an empty ``BsonDocument`` as a query filter.

   To see a fully runnable example of using the ``Find()`` method to find all documents, see 
   :ref:`csharp-retrieve-additional-information`.   
```

Reply:
Good suggestion